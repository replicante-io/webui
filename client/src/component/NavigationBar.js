'use strict';
//@flow

import { Component } from 'react';
import React from 'react';

import { NavLink } from 'react-router-dom';


const NAVBAR_TOGGLE_TAG = 'navbarNavAltMarkup';


class NavigationBar extends Component<{}> {
  renderItems() {
    return (
      <div className="collapse navbar-collapse" id={NAVBAR_TOGGLE_TAG}>
        <div className="navbar-nav">
          <NavLink className="nav-item nav-link" to="/" exact={true}>Overview</NavLink>
          <NavLink className="nav-item nav-link" to="/clusters">Clusters</NavLink>
          <NavLink className="nav-item nav-link" to="/events">Events</NavLink>
          <NavLink className="nav-item nav-link" to="/automations">Automations</NavLink>
          <NavLink className="nav-item nav-link" to="/admin">Admin</NavLink>
        </div>
      </div>
    );
  }

  renderToggleButton() {
    return (
      <button className="navbar-toggler" type="button" data-toggle="collapse"
              data-target={"#" + NAVBAR_TOGGLE_TAG} aria-controls={NAVBAR_TOGGLE_TAG}
              aria-expanded="false" aria-label="Toggle navigation">
        <span className="navbar-toggler-icon"></span>
      </button>
    );
  }

  render() {
    return (
      <nav className="navbar sticky-top navbar-expand-sm navbar-dark bg-dark">
        {this.renderToggleButton()}
        {this.renderItems()}
      </nav>
    );
  }
}


export default NavigationBar;
