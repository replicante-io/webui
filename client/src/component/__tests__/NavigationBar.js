'use strict';
//@flow

import { BrowserRouter } from 'react-router-dom';
import React from 'react';
import renderer from 'react-test-renderer';

import NavigationBar from '../NavigationBar';


describe('NavigationBar', () => {
  test('renders correctly', () => {
    const tree = renderer.create(
      <BrowserRouter>
        <NavigationBar />
      </BrowserRouter>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
