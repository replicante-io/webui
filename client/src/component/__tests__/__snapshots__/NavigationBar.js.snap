// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationBar renders correctly 1`] = `
<nav
  className="navbar sticky-top navbar-expand-sm navbar-dark bg-dark"
>
  <button
    aria-controls="navbarNavAltMarkup"
    aria-expanded="false"
    aria-label="Toggle navigation"
    className="navbar-toggler"
    data-target="#navbarNavAltMarkup"
    data-toggle="collapse"
    type="button"
  >
    <span
      className="navbar-toggler-icon"
    />
  </button>
  <div
    className="collapse navbar-collapse justify-content-center"
    id="navbarNavAltMarkup"
  >
    <div
      className="navbar-nav"
    >
      <a
        aria-current={false}
        className="nav-item nav-link"
        href="/clusters"
        onClick={[Function]}
        style={undefined}
      >
        Clusters
      </a>
      <a
        aria-current={false}
        className="nav-item nav-link"
        href="/events"
        onClick={[Function]}
        style={undefined}
      >
        Events
      </a>
      <a
        aria-current={false}
        className="nav-item nav-link"
        href="/"
        onClick={[Function]}
        style={undefined}
      >
        Overview
      </a>
      <a
        aria-current={false}
        className="nav-item nav-link"
        href="/automations"
        onClick={[Function]}
        style={undefined}
      >
        Automations
      </a>
      <a
        aria-current={false}
        className="nav-item nav-link"
        href="/admin"
        onClick={[Function]}
        style={undefined}
      >
        Admin
      </a>
    </div>
  </div>
</nav>
`;
